### PR TITLE
DEMO 426 track file service refactor

### PR DESCRIPTION
**Pull request by kiley-mitti**
[DEMO-426](https://rocketcom.atlassian.net/browse/DEMO-426)

## Brief Description

This refactors the tracking data utility components to use the tracking data utility service to handle their table state.  This is so that when you return to 'Select Files' the filter and selection state will be in the same condition it was in before. When a new satellite is selected the table data filtering and selection resets.

## Final Checklist

- [x] Works in Chrome
- [x] Works in Firefox
- [x] Works in Safari
- [x] Handles responsiveness as required
- [x] Dark theme styles are correct
- [x] Light theme styles are correct

### CODE REVIEW

File: src/app/main/utility-components/track-data-utility/track-data/track-data-graph/track-data-graph.component.html [hunk 1]
```diff
@@ -20,32 +20,39 @@
...
+    <span>OD Quality Est: 82/100</span>
+    <span>Data Points: {{ dataPointLength }}</span>
+  </footer>
+</ng-container>
+<ng-template #noDataTemplate>
+  <div class="no-files">
+    <h2>No Track Files Selected</h2>
+    <p>Please select track files to view the chart.</p>
+  </div>
+</ng-template>
```
Comment (NicDominguez) [inline src/app/main/utility-components/track-data-utility/track-data/track-data-graph/track-data-graph.component.html:58]: Good call

File: src/app/main/utility-components/track-data-utility/track-data/track-data-graph/track-data-graph.component.ts [hunk 2]
```diff
@@ -175,9 +167,11 @@ export class TrackDataGraphComponent {
...
     const updatedSeries = this.series.filter((data: any) => {
-      if (data.visible) {
-        markers.push(data.markerSize);
-        dataPoints = dataPoints + data.data.length;
+      if (data.visible){ markers.push(data.markerSize);
+        dataPoints = dataPoints + data.data.length
+      }
+      if (data.visible){ markers.push(data.markerSize);
+        dataPoints = dataPoints + data.data.length
       }
```
Comment (NicDominguez) [inline src/app/main/utility-components/track-data-utility/track-data/track-data-graph/track-data-graph.component.ts:175]: duplication mistake?

**Review by NicDominguez — CHANGES_REQUESTED** [review]
Just one duplicate code section that I think is an error.

**Review by NicDominguez — CHANGES_REQUESTED** [review]
Looks like there might be a styling error with the edit track file container. 
image

File: src/app/main/utility-components/track-data-utility/track-data/track-data-graph/track-data-graph.component.ts (hunk 2 above)
Comment (kiley-mitti) [reply src/app/main/utility-components/track-data-utility/track-data/track-data-graph/track-data-graph.component.ts]: yup! I find the merge editor in vscode to be a bit confusing. I'm getting better at dealing with it but... sometimes that happens :) Fixed!

**Comment by kiley-mitti** [comment]
> Looks like there might be a styling error with the edit track file container.
> image

Turns out this is a symptom of a larger issue. Adding the child tab component here completely changed when these components are created and destroyed and in doing so broke the chart updating. :) I'll be back with you when it is fixed.

**Comment by kiley-mitti** [comment]
ok! assuming this compiles I do believe those bugs to be fixed :)

File: src/app/main/utility-components/track-data-utility/track-files-data.service.ts [hunk 3]
```diff
@@ -1,30 +1,153 @@
...
+    columnDefs: ColumnDefs<TrackData>[] = [
+      { header: '', field: 'id' },
+      { header: 'File Name', field: 'name', sortable: true },
+      { header: 'Date', field: 'creationDate', sortable: true },
+      { header: 'File Size', field: 'fileSize', sortable: true },
+    ];
+
+    // Cleans up subscriptions to avoid memory leaks
+    destroyRef = inject(DestroyRef);
+    destroyed = new Subject();
```
Comment (FMorrison87) [inline src/app/main/utility-components/track-data-utility/track-files-data.service.ts:40]: I actually don't think you need to add this if you're using `takeUntilDestroyed()`, same for the code in `ngOnDestroy()`

Comment (FMorrison87) [reply src/app/main/utility-components/track-data-utility/track-files-data.service.ts:40]: https://indepth.dev/posts/1518/takeuntildestroy-in-angular-v16

**Review by FMorrison87 — APPROVED** [review]

**Review by NicDominguez — APPROVED** [review]

**Pull request merged**